src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -271,15 +318,13 @@ public void RunClassInitialize(TestContext testContext)
             var realException = this.ClassInitializationException.InnerException ?? this.ClassInitializationException;
 
             var outcome = UnitTestOutcome.Failed;
-            string errorMessage = null;
-            StackTraceInformation exceptionStackTraceInfo = null;
-            if (!realException.TryGetUnitTestAssertException(out outcome, out errorMessage, out exceptionStackTraceInfo))
+            if (!realException.TryGetUnitTestAssertException(out outcome, out string errorMessage, out StackTraceInformation exceptionStackTraceInfo))
             {
                 errorMessage = string.Format(
                     CultureInfo.CurrentCulture,
                     Resource.UTA_ClassInitMethodThrows,
                     this.ClassType.FullName,
-                    this.ClassInitializeMethod.Name,
+                    this.ClassInitializeMethod?.Name ?? initializeMethod.Name,
```

Comment:
hmm, we might want to store the method that failed and report the name here. This would report the childs method failed even if the actual failure happened in one of the base classes.

Reply:
yup, that's true. missed that part. Created a local variable to hold the failed initialize method name.